frontend/src/infra/MyProvider.js

```diff
@@ -1,143 +1,107 @@
-import React, { useState } from 'react';
+import React, { useReducer, useEffect } from 'react';
+import { useHistory } from 'react-router-dom';
+import axios from 'axios';
+
 import { MyContext } from './MyContext';
 import {
-  PHASE,
   SERVER_ADDRESS,
   SERVER_PORT,
   ROOM_ROUTES,
 } from '../shared_code/consts';
-import Shortid from 'shortid';
-import axios from 'axios';
-import { createSocketAndListen as createRoomSocket } from './socketUtils.js';
+import { verifySocketListen } from './socketUtils';
+
+import { playerReducer, playerInitialState } from './PlayerReducer';
+import { roomInitialState, roomReducer } from './RoomReducer';
 
 const MyProvider = (props) => {
-  const [state, setState] = useState({
-    phase: PHASE.WELCOME_SCREEN,
-    currentRoom: '',
-    isUserEnteredName: false,
-    playerName: '',
-    playerId: '',
-    roomPlayersList: [],
-    joinDate: '',
-    roomCreationDate: '',
-    roomSocket: null,
-    roomSocketPort: null,
-  });
+  const [player, playerDispatch] = useReducer(
+    playerReducer,
+    playerInitialState,
+  );
 
-  const onRoomCreated = () => {
-    const playerId = Shortid.generate();
-    setState({
-      ...state,
-      phase: PHASE.WAITING_ROOM,
-      playerId,
+  const [room, roomDispatch] = useReducer(roomReducer, roomInitialState);
+
+  const updatePlayerList = (data) => {
+    roomDispatch({
+      type: 'updatePlayerList',
```

Comment:
make this a const, and put it outside this file

Reply:
fe277e0

Reply:
4f14963198d17ff381fe702ef67f4589a5e548a8